velox/connectors/hive/HiveDataSink.cpp

```diff
@@ -502,7 +502,7 @@ uint32_t HiveDataSink::appendWriter(const HiveWriterId& id) {
           writePath,
           {.bufferWrite = false,
            .connectorProperties = connectorProperties_,
-           .pool = connectorQueryCtx_->memoryPool(),
+           .pool = connectorQueryCtx_->connectorMemoryPool(),
```

Comment:
Connector memory pool is aggregate pool. Meta internal file sink expect leaf pool. Thanks!

Reply:
@xiaoxmeng I modified the FileOptions and FileSink::options API to consume a leaf memory pool. Can you take a look?